app/javascript/controllers/image_upload_controller.js

```diff
@@ -1,56 +1,80 @@
 import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
-  static targets = ["file", "content"]
+  static targets = [ "file", "content", "preview" ]
 
   connect() {
-    this.dropHandler()
-    this.pasteHandler()
+    this.fileTarget.addEventListener("change", this.boundPreviewUpdate)
+    this.contentTarget.addEventListener("dragover", this.boundIgnore)
+    this.contentTarget.addEventListener("dragenter", this.boundIgnore)
+    this.contentTarget.addEventListener("drop", this.boundDrop)
+    this.contentTarget.addEventListener("paste", this.boundPasteHandler)
   }
 
-  dropHandler() {
-    this.fileTarget.addEventListener("dragover", (event) =>
-      this.preventDefaults(event),
-    )
-    this.fileTarget.addEventListener("dragenter", (event) =>
-      this.preventDefaults(event),
-    )
-    this.fileTarget.addEventListener("drop", (event) => this.drop(event))
+  disconnect() {
+    this.fileTarget.removeEventListener("change", this.boundPreviewUpdate)
+    this.contentTarget.removeEventListener("dragover", this.boundIgnore)
+    this.contentTarget.removeEventListener("dragenter", this.boundIgnore)
+    this.contentTarget.removeEventListener("drop", this.boundDrop)
+    this.contentTarget.removeEventListener("paste", this.boundPasteHandler)
+  }
+
+  boundPreviewUpdate = () => { this.previewUpdate() }
+  previewUpdate() {
+    const input = this.fileTarget
+    if (input.files && input.files[0]) {
+      const reader = new FileReader()
+      reader.onload = (e) => {
+        this.previewTarget.querySelector("img").src = e.target.result
+        this.element.classList.add("show-previews")
+        this.contentTarget.focus()
+        window.dispatchEvent(new Event('resize')) // Throw this event will cause textarea_autogrow to reprocess
+      }
+      reader.readAsDataURL(input.files[0])
```

Comment:
This is cool. I had no idea FileReader existed.

Reply:
ChatGPT taught me that one. :)